packages/cli/src/commands/init/templateName.js

```diff
@@ -0,0 +1,48 @@
+// @flow
+import path from 'path';
+import {URL} from 'url';
+import {fetch} from '../../tools/fetch';
+
+const FILE_PROTOCOL = /file:/;
+
+function handleFileProtocol(filePath: string) {
+  const uri = new URL(filePath).pathname;
+
+  return {
+    uri,
+    name: require(path.join(uri, 'package.json')).name,
+  };
+}
+
+export function processTemplateName(templateName: string) {
+  if (templateName.match(FILE_PROTOCOL)) {
+    return handleFileProtocol(templateName);
+  }
+
+  return {
+    uri: templateName,
+    name: templateName,
+  };
+}
+
+/**
+ * `init` may be invoked with a shorthand like `--template typescript`
+ * which should resolve to `react-native-template-typescript` package.
+ * To support that, we query npm registry if a package like this exists, if not
+ * we return the original name without a change.
+ */
+export async function tryTemplateShorthand(maybeTemplateShorthand: string) {
+  try {
+    const reactNativeTemplatePackage = `react-native-template-${maybeTemplateShorthand}`;
+    const response = await fetch(
```

Comment:
I don't think it's good to make an HTTP request to the registry with the local string here. Can we first check if it is a local path (has path.separator in it) or whether it has `file://` prefixed?